test/topicFilters.spec.ts

```diff
@@ -0,0 +1,296 @@
+  try {
+    const rules = await client.getRules();
+    for (let i = 0; i < rules.length; i++) {
+      const rule = rules[i];
+      await client.removeRule(rule.name);
+    }
+  } catch (err) {
+    console.log("Error while removing the rule", err);
+  }
+}
+
+let namespace: Namespace;
+let subscriptionClient: SubscriptionClient;
+let defaultSubscriptionClient: SubscriptionClient;
+
+async function beforeEachTest(): Promise<void> {
+  // The tests in this file expect the env variables to contain the connection string and
+  // the names of empty queue/topic/subscription that are to be tested
+
+  if (!process.env.SERVICEBUS_CONNECTION_STRING) {
+    throw new Error(
+      "Define SERVICEBUS_CONNECTION_STRING in your environment before running integration tests."
+    );
+  }
+  if (!process.env.TOPIC_NAME) {
+    throw new Error("Define TOPIC_NAME in your environment before running integration tests.");
+  }
+  if (!process.env.SUBSCRIPTION_NAME) {
+    throw new Error(
+      "Define SUBSCRIPTION_NAME in your environment before running integration tests."
+    );
+  }
+  if (!process.env.DEFAULT_SUBSCRIPTION_NAME) {
+    throw new Error(
+      "Define DEFAULT_SUBSCRIPTION_NAME in your environment before running integration tests."
+    );
+  }
+
+  namespace = Namespace.createFromConnectionString(process.env.SERVICEBUS_CONNECTION_STRING);
+  subscriptionClient = namespace.createSubscriptionClient(
+    process.env.TOPIC_NAME,
+    process.env.SUBSCRIPTION_NAME
+  );
+  defaultSubscriptionClient = namespace.createSubscriptionClient(
+    process.env.TOPIC_NAME,
+    process.env.DEFAULT_SUBSCRIPTION_NAME
+  );
+  await removeAllRules(subscriptionClient);
+}
+
+async function afterEachTest(): Promise<void> {
+  await namespace.close();
+}
+
+describe("Topic Filters - Tests", function(): void {
+  beforeEach(async () => {
+    await beforeEachTest();
+  });
+
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  it("Add Rule with Boolean filter", async function(): Promise<void> {
+    await subscriptionClient.addRule("TrueFilter", true);
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "TrueFilter");
+  });
+
+  it("Add Rule with SQL filter", async function(): Promise<void> {
+    await subscriptionClient.addRule(
+      "Priority_1",
+      "(priority = 1 OR priority = 2) AND (sys.label LIKE '%String2')"
+    );
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Priority_1");
+  });
+
+  it("Add Rule with SQL filter and action", async function(): Promise<void> {
+    await subscriptionClient.addRule(
+      "Priority_1",
+      "(priority = 1 OR priority = 3) AND (sys.label LIKE '%String1')",
+      "SET sys.body = 'MessageX'"
+    );
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Priority_1");
+  });
+
+  it("Add Rule with SQL filter and action", async function(): Promise<void> {
```

Comment:
This test case is same as the one before it

Reply:
Oops! I was testing the action part in this temporary code and forgot to remove this.